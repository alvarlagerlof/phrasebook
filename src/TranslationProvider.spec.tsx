import React from 'react';

import { render } from '@testing-library/react';
import { TranslationProvider, useTranslation } from './TranslationProvider';
import { TranslationArguments } from './types';

describe('TranslationProvider', () => {
  const locale = 'en-GB';
  const translations = {
    search: {
      label: 'Search',
    },
    reviews: '{{count}} review',
    reviews_plural: '{{count}} reviews',
    boardBasis: {
      code: 'Board Basis',
      code_AI: 'All Inclusive',
    },
  };

  it('translates using useTranslation hook', () => {
    const Test: React.FC<{ name: string; args?: TranslationArguments }> = ({ name, args }) => {
      const { t } = useTranslation();

      return (
        <span>{t(name, args)}</span>
      );
    };

    const { asFragment } = render(
      <TranslationProvider
        locale={locale}
        translations={translations}
      >
        <Test
          name="search.label"
        />
        <Test
          name="reviews"
          args={{ count: 1 }}
        />
        <Test
          name="reviews"
          args={{ count: 4 }}
        />
        <Test
          name="boardBasis.code"
        />
        <Test
          name="boardBasis.code"
          args={{ context: 'AI' }}
        />
      </TranslationProvider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
