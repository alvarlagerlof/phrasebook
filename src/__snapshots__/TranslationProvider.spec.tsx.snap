// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TranslationProvider translates using useTranslation hook 1`] = `
<DocumentFragment>
  <span>
    Search
  </span>
  <span>
    1 review
  </span>
  <span>
    4 reviews
  </span>
  <span>
    Board Basis
  </span>
  <span>
    All Inclusive
  </span>
</DocumentFragment>
`;
